### PR TITLE
Add functionality for bcc/cc

### PR DESCRIPTION
**Pull request by d-m-u**
We have bcc/cc functionality in the main repo (sans tests) and we could be using it in automate too if it were exposed so this is that. 

It's for this: https://bugzilla.redhat.com/show_bug.cgi?id=1688500

### CODE REVIEW

**Comment by d-m-u** [comment]
I guess I almost think these fields should be nil by default, @tinaafitz

**Comment by d-m-u** [comment]
@miq-bot remove_label wip

File: lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb [hunk 1]
```diff
@@ -7,12 +7,17 @@ class MiqAeServiceMethods
 
     SYNCHRONOUS = Rails.env.test?
 
-    def self.send_email(to, from, subject, body, content_type = nil)
+    def self.send_email(to, from, subject, body, args = {})
       ar_method do
         meth = SYNCHRONOUS ? :deliver : :deliver_queue
+        content_type = args[:content_type] || nil
```
Comment (jrafanie) [inline lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb]: What if they provide the `content_type`?

Comment (jrafanie) [reply lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb]: What was content_type before when it was non-nil?

File: lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method.rb [hunk 2]
```diff
@@ -34,7 +34,7 @@ def self.miq_log_workspace(obj, _inputs)
     end
 
     def self.miq_send_email(_obj, inputs)
-      MiqAeMethodService::MiqAeServiceMethods.send_email(inputs["to"], inputs["from"], inputs["subject"], inputs["body"])
+      MiqAeMethodService::MiqAeServiceMethods.send_email(inputs["to"], inputs["from"], inputs["cc"], inputs["bcc"], inputs["subject"], inputs["body"])
```
Comment (lfu) [inline lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method.rb]: This seems not matching ```send_email``` method signature.

Comment (jrafanie) [reply lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method.rb]: Do we ever use the optional content_type?  Is this the only caller?

File: lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb [hunk 3]
```diff
@@ -7,12 +7,17 @@ class MiqAeServiceMethods
 
     SYNCHRONOUS = Rails.env.test?
 
-    def self.send_email(to, from, subject, body, content_type = nil)
+    def self.send_email(to, from, subject, body, args = {})
```
Comment (lfu) [inline lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_methods.rb]: This method is being used only in automate. I don't see any caller of the method that specifies ```content_type```.

The  method signature can be changed to:
```
def self.send_mail(to, from, subject, body, content_type: nil, cc: nil, bcc: nil)
```

**Comment by lfu** [comment]
@jrafanie Please review.

**Comment by d-m-u** [comment]
@lfu he's on pto this week

File: lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method.rb [hunk 4]
```diff
@@ -34,7 +34,7 @@ def self.miq_log_workspace(obj, _inputs)
     end
 
     def self.miq_send_email(_obj, inputs)
-      MiqAeMethodService::MiqAeServiceMethods.send_email(inputs["to"], inputs["from"], inputs["subject"], inputs["body"])
+      MiqAeMethodService::MiqAeServiceMethods.send_email(inputs["to"], inputs["from"], inputs["subject"], inputs["body"], inputs["cc"], inputs["bcc"])
```
Comment (lfu) [inline lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method.rb]:
```
 MiqAeMethodService::MiqAeServiceMethods.send_email(inputs["to"], inputs["from"], inputs["subject"], inputs["body"], :cc => inputs["cc"], :bcc => inputs["bcc"], :content_type => inputs["content_type"])
```

Comment (d-m-u) [reply lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_builtin_method.rb]: What's the point of using positional args if you're specifying which one they are here anyway?

**Comment by d-m-u** [comment]
@miq-bot add_reviewer @gmcculloug

**Comment by d-m-u** [comment]
Can I please get people to look at this? It's wanted for upcoming build.

**Comment by lfu** [comment]
@d-m-u Could you please remove the WIP in the title?

**Comment by d-m-u** [comment]
@miq-bot remove_label wip

**Comment by d-m-u** [comment]
@lfu could you approve this, please?

**Comment by simaishi** [comment]
@d-m-u is this `hammer/yes`?

**Comment by d-m-u** [comment]
@miq-bot add_label hammer/yes

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit e94b528ab2f9257c78ddabfda43947887ac85ff8
Author: tina <tina.fitzgerald2@gmail.com>
Date:   Thu Apr 18 15:20:11 2019 -0400

    Merge pull request #313 from d-m-u/adding_bcc_and_cc_functionality
    
    Add functionality for bcc/cc
    
    (cherry picked from commit 34ca401a667233c49f4952c78cf4a43883837ac6)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1702304
```

**Pull request merged**